### PR TITLE
Added google cloud build 

### PR DESCRIPTION
**Pull request by MJafarMashhadi**
branch name is misleading

### CODE REVIEW

File: cloudbuild.yaml [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+         '--file', 'Dockerfile',
+         '.']
+  id: 'step-build-base'
+
+# Step2: Build the production image by copying
+# the source code into the image
+#
+- name: 'gcr.io/cloud-builders/docker'
+  args: ['build',
+         '--tag', 'us.gcr.io/$PROJECT_ID/alchemy:latest',
```
Comment (MJafarMashhadi) [inline cloudbuild.yaml:18]: Need to change it to `gcr.io`

File: alchemy/admin_server/external_services.py [hunk 2]
```diff
@@ -1,16 +1,21 @@
...
-
 
 class GCPPubSubService:
-    publish_client = pubsub.PublisherClient()
+    _client = None
+
+    @classmethod
+    def get_client(cls):
+        if cls._client is None:
+            from google.cloud import pubsub
```
Comment (ShahNewazKhan) [inline alchemy/admin_server/external_services.py:10]: Moving the import to outside and at the top of the file will prevent from it being loaded each time the method is called.

File: cloudbuild.yaml [hunk 3]
```diff
@@ -0,0 +1,36 @@
+steps:
+# Step1: Build the alchemy development image
+# that does not include the source codes as
+# the base
+#
+- name: 'gcr.io/cloud-builders/docker'
+  args: ['build',
+         '--tag', 'alchemy-base:latest',
```
Comment (ShahNewazKhan) [inline cloudbuild.yaml:8]: We should use the git commit hash to tag the image as it makes it easier to debug later without having to generate the image digest and doing a reverse look up in the gcr. 

Cloud build provides both `$COMMIT_SHA` and `$SHORT_SHA` as detailed here https://cloud.google.com/cloud-build/docs/configuring-builds/substitute-variable-values

File: alchemy/admin_server/external_services.py (hunk 2 above)
Comment (MJafarMashhadi) [reply alchemy/admin_server/external_services.py:10]: This is a lazy loading function, the import will be called only once in the if statement. Moved it to inside the method to have a cleaner namespace, however it's not a crucial part of my PR, I can move them back on top of the file if it's better.

File: cloudbuild.yaml (hunk 3 above)
Comment (MJafarMashhadi) [reply cloudbuild.yaml:8]: Thanks. The image label was a kind of grey area for me and I was looking forward to a feedback. So would a label like `dev-$SHORT_SHA` or `test-$SHORT_SHA` be good?

File: alchemy/admin_server/external_services.py (hunk 2 above)
Comment (jzhang-gp) [reply alchemy/admin_server/external_services.py:10]: Generally I would prefer all the imports at the top so it's consistent and not scattered around from a readability perspective ( Java developer bias here ;) )

File: cloudbuild.yaml (hunk 3 above)
Comment (ShahNewazKhan) [reply cloudbuild.yaml:8]: No worries! Generally the idea is to build one image to be used through out all environments so just `$SHORT_SHA` is good.

Comment (MJafarMashhadi) [reply cloudbuild.yaml:8]: Makes sense, Thanks!

**Review by ShahNewazKhan — APPROVED** [review]
LGTM!

**Review by jzhang-gp — APPROVED** [review]
LGTM

**Pull request merged**